danascheider/skyrim_inventory_management_frontend
# Display inventory lists on inventory lists page

## Context

[**Display inventory lists on inventory lists page**](https://trello.com/c/68t9wbtZ/145-display-inventory-lists-on-inventory-lists-page)

In the [previous PR](https://github.com/danascheider/skyrim_inventory_management_frontend/pull/131), we created a new `InventoryPage` component and route. The next step is to make this page display inventory lists for the selected game. As part of this work, I also enabled users to create games using the dropdown or the "Create a game" link (if there are no games already).

## Changes

Summary of the changes made, preferably as a bulleted list

## Considerations

* Create `InventoryList`, `InventoryListItem`, and `InventoryPageContent` components to display inventory lists
* Add modal to `InventoryPage` to enable game creation using the modal form
* Add `fetchInventoryLists` function to the `simApi` module to fetch inventory lists for the specified game
* Create `InventoryListsContext` that is responsible for maintaining the list of inventory lists and their items
* Add tests for displaying inventory lists and creating games from the inventory lists page
* Add Storybook stories for new components
* Update and build out Storybook stories for the `InventoryPage` component
* Add `unit weight` field to shopping list item component 

## Manual Test Cases

The new inventory list page is similar to the shopping lists page, although it doesn't have the same functionality yet. Start by removing all games from your development database and verifying that you can create a game through the games dropdown on the inventory page (by typing a name in the input and pressing enter). Then, delete that game, visit the inventory page again, and click the "Create a Game" button to verify you can create a game through the link. If you enter an invalid name, you should see a validation error.

Your ability to test further may be limited by the fact that inventory lists can't be created through the UI yet.

## Screenshots and GIFs

### InventoryList Component

<img width="637" alt="Screen Shot 2021-08-14 at 9 56 48 am" src="https://user-images.githubusercontent.com/5115928/129428453-de9accd0-54f0-4b17-9d54-acf23e254ca0.png">

### InventoryList Component with Expanded List Item

<img width="629" alt="Screen Shot 2021-08-14 at 9 56 58 am" src="https://user-images.githubusercontent.com/5115928/129428460-4fbd6db3-954e-409d-8969-9db21fe7f275.png">

### InventoryPage

<img width="1167" alt="Screen Shot 2021-08-14 at 9 57 36 am" src="https://user-images.githubusercontent.com/5115928/129428499-413c673f-8aa6-42fc-8537-a09d3e0fd5f9.png">

### InventoryPage - No Games

<img width="928" alt="Screen Shot 2021-08-14 at 10 13 34 am" src="https://user-images.githubusercontent.com/5115928/129428521-ddb5364d-3f7e-428f-ac14-df1fdcd55a17.png">

### InventoryPage - No Lists

<img width="927" alt="Screen Shot 2021-08-14 at 10 17 23 am" src="https://user-images.githubusercontent.com/5115928/129428653-bf40302a-9510-4c29-9625-794f91c457e8.png">

### InventoryPage - Loading Error

<img width="928" alt="Screen Shot 2021-08-14 at 10 14 41 am" src="https://user-images.githubusercontent.com/5115928/129428563-fc574a11-0a19-45aa-924d-8fd7cd58535a.png">
